cpp/tests/experimental/spatial/pairwise_point_in_polygon_test.cu

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuspatial/error.hpp>
+#include <cuspatial/experimental/iterator_factory.cuh>
+#include <cuspatial/experimental/pairwise_point_in_polygon.cuh>
+#include <cuspatial/vec_2d.hpp>
+
+#include <rmm/device_vector.hpp>
+
+#include <thrust/host_vector.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/iterator/transform_iterator.h>
+
+#include <gtest/gtest.h>
+
+using namespace cuspatial;
+
+template <typename T>
+struct PairwisePointInPolygonTest : public ::testing::Test {
+ public:
+  rmm::device_vector<vec_2d<T>> make_device_points(std::initializer_list<vec_2d<T>> pts)
+  {
+    return rmm::device_vector<vec_2d<T>>(pts.begin(), pts.end());
+  }
+
+  rmm::device_vector<std::size_t> make_device_offsets(std::initializer_list<std::size_t> pts)
+  {
+    return rmm::device_vector<std::size_t>(pts.begin(), pts.end());
+  }
+};
+
+// float and double are logically the same but would require separate tests due to precision.
+using TestTypes = ::testing::Types<float, double>;
+TYPED_TEST_CASE(PairwisePointInPolygonTest, TestTypes);
+
+TYPED_TEST(PairwisePointInPolygonTest, OnePolygonOneRing)
+{
+  using T                = TypeParam;
+  auto point_list        = std::vector<std::vector<T>>{{-2.0, 0.0},
+                                                {2.0, 0.0},
+                                                {0.0, -2.0},
+                                                {0.0, 2.0},
+                                                {-0.5, 0.0},
+                                                {0.5, 0.0},
+                                                {0.0, -0.5},
+                                                {0.0, 0.5}};
+  auto poly_offsets      = this->make_device_offsets({0});
+  auto poly_ring_offsets = this->make_device_offsets({0});
+  auto poly_point =
+    this->make_device_points({{-1.0, -1.0}, {1.0, -1.0}, {1.0, 1.0}, {-1.0, 1.0}, {-1.0, -1.0}});
+
+  auto got      = rmm::device_vector<int32_t>(1);
+  auto expected = std::vector<bool>{false, false, false, false, true, true, true, true};
```

Comment:
Avoid `vector<bool>`. https://isocpp.org/blog/2012/11/on-vectorbool

In tests where you compare against device data, just use a `thrust::device_vector` or `thrust::host_vector`.

This may be why these tests are failing in CI?

Reply:
```suggestion
  auto expected = thrust::host_vector<bool>{false, false, false, false, true, true, true, true};
```

Reply:
yikes!

Reply:
`thrust::host_vector` doesn't have the same constructor options as `std::vector`, I can't pass in a list of bools. How do you feel about me using `std::vector<int>` instead?

Reply:
Using `std::vector<int>`

Reply:
Issue created for this: https://github.com/rapidsai/cuspatial/issues/823